route_info/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+use geom::{Distance, FindClosest, LonLat};
+use map_model::{IntersectionID, Map};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct RouteInfo {
+    map: Map,
+    closest_intersection: FindClosest<IntersectionID>,
+}
+
+#[wasm_bindgen]
+impl RouteInfo {
+    #[wasm_bindgen(constructor)]
+    pub fn new(map_bytes: &[u8]) -> Result<RouteInfo, JsValue> {
+        // Panics shouldn't happen, but if they do, console.log them.
+        console_error_panic_hook::set_once();
+
+        web_sys::console::log_1(&format!("Got {} bytes, deserializing", map_bytes.len()).into());
+
+        let map: Map = bincode::deserialize(map_bytes).map_err(err_to_js)?;
+
+        let mut closest_intersection = FindClosest::new();
+        for i in map.all_intersections() {
+            closest_intersection.add_polygon(i.id, &i.polygon);
+        }
+
+        Ok(RouteInfo {
+            map,
+            closest_intersection,
+        })
+    }
+
+    // TODO How can we glue up anyhow?
+    // TODO Can we take geojson::Feature so that it shows up in TS?
+    #[wasm_bindgen(js_name = nameForRoute)]
+    pub fn name_for_route(&self, raw_line_string: JsValue) -> Result<String, JsValue> {
+        let feature: geojson::Feature = serde_wasm_bindgen::from_value(raw_line_string)?;
+        // TODO Duplicate non-adjacent points; make a more permissive version of PolyLine.
+        //let polyline = PolyLine::from_geojson(&feature, Some(self.map.get_gps_bounds())).map_err(err_to_js)?;
```

Comment:
A/B Street's geometry library has methods to much more easily parse GeoJSON, but they're too strict about duplicate points. We could improve this later to simplify code here.